src/pages/Signup/index.jsx

```diff
@@ -1,10 +1,31 @@
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import styled from "styled-components";
 import { BackIcon } from "../../assets/icons";
 import { Button, Input } from "../../components";
 import hero from "../../assets/backgrounds/authBackground.png";
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { signUp } from "../../features/auth/authSlice";
 
 export function Signup() {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
```

Comment:
Suggestion:- Instead of using different useStates, You can set all the details in an object and pass it into a single useState.